ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala

```diff
@@ -44,7 +44,11 @@ object IntegrityCheck extends App {
   val ledgerDumpStream: DataInputStream =
     new DataInputStream(new FileInputStream(filename))
 
-  val engine = Engine()
+  // FIXME: https://github.com/digital-asset/daml/issues/5164
```

Comment:
If the dev version may produce different transactions from the stable one the integrity checker must use the latest stable version otherwise we'd get mismatches when checking against the backed up ledger entries.

Reply:
I will take care of that in an upcoming PR. For now and until #5164 is completed the versioning of LF/transaction/version is frozen, so no risk of issue there.